packages/components/src/ui/tooltip/component.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { contextConnect, useContextSystem } from '@wp-g2/context';
+import { mergeRefs } from '@wp-g2/utils';
+// eslint-disable-next-line no-restricted-imports
+import { TooltipReference, useTooltipState } from 'reakit';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo, cloneElement } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { TooltipContext } from './context';
+import TooltipContent from './content';
+import { TooltipShortcut } from './styles';
+
+/**
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<import('./types').Props, 'div'>} props
+ * @param {import('react').Ref<any>} forwardedRef
+ */
+function Tooltip( props, forwardedRef ) {
+	const {
+		animated = true,
+		animationDuration = 160,
+		baseId,
+		children,
+		content,
+		focusable = true,
+		gutter = 4,
+		id,
+		modal = true,
+		placement,
+		visible = false,
+		shortcut,
+		...otherProps
+	} = useContextSystem( props, 'Tooltip' );
+
+	const tooltip = useTooltipState( {
+		animated: animated ? animationDuration : undefined,
+		baseId: baseId || id,
+		gutter,
+		placement,
+		visible,
+		...otherProps,
+	} );
+
+	const contextProps = useMemo(
+		() => ( {
+			tooltip,
+		} ),
+		[ tooltip ]
+	);
+
+	// @todo(itsjonq) Does `ref` ever really exist on children? What's the use case for this? Why not just pass the forwarded ref? Isn't this just going to unexpectedly swap out the ref from the intended element to another?
```

Comment:
@ItsJonQ Please see the note here

Reply:
@sarayourfriend Ah! Nice catch. I think this was a result of a (so-so) refactor to achieve the ["abstracted example" from Reakit](https://reakit.io/docs/tooltip/#abstracting).

After poking at it via Storybook, I think the setup should be something like:

```jsx
	return (
		<TooltipContext.Provider value={ contextProps }>
			{ content && (
				<TooltipContent unstable_portal={ modal } ref={forwardedRef}>
					{ content }
					{ shortcut && <TooltipShortcut shortcut={ shortcut } /> }
				</TooltipContent>
			) }
			{ children && (
				<TooltipReference
					{ ...tooltip }
					{ ...children.props }
					ref={ children.ref }
				>
					{ ( referenceProps ) => {
						if ( ! focusable ) {
							referenceProps.tabIndex = undefined;
						}
						return cloneElement( children, referenceProps );
					} }
				</TooltipReference>
			) }
		</TooltipContext.Provider>
	);
```

The `forwardedRef` goes to the `TooltipContent`, and the trigger's ref stays in place.
That way, you can have access to the `TooltipContent` node, if needed.

```jsx

export const _default = () => {
	const ref = useRef()
	const tooltipContentRef = useRef()

	return (
		<Tooltip
			ref={tooltipContentRef}
		>
			<Text ref={ref}>Hello</Text>
		</Tooltip>
	);
};
```